modules/recon/dns/dns.go

```diff
@@ -0,0 +1,344 @@
+package dns
+
+import (
+	"encoding/gob"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/miekg/dns"
+	"github.com/netm4ul/netm4ul/core/config"
+	"github.com/netm4ul/netm4ul/core/server/database"
+	"github.com/netm4ul/netm4ul/modules"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// DnsResult represent the parsed ouput
+type DnsResult struct {
+	Types    map[string][]string
+	resolver string
+	err      string
```

Comment:
Resolver ?
Err ?
If this is going to be serialized, fields must be set as Public

Reply:
done